### DISCUSSION TITLE
WOL from an external HTTP request

### DISCUSSION THREAD

**Discussion by dupuysylvain** [seriousm4x/UpSnap#172] [Help wanted] [answered]
Hello,

I was wondering if it was possible to use an HTTP request to WOL one of the PCs ?

Thanks,
Dupuy Sylvain

**Comment by seriousm4x** [comment, marked as answer]
Hey, yes it is possible. You can use `/api/upsnap/wake/:id`.
See the wiki here: https://github.com/seriousm4x/UpSnap/wiki/Rest-API#ive-extended-the-default-pocketbase-api

Make sure to authorize first and then send the token as bearer.

**Reply by dupuysylvain** [reply]
Amazing thanks for your quick response,  should have checked the wiki

**Reply by JonasHRR** [reply]
How do I authorize and/or send a token? Can't I just disable that somehow? I just want to use a link to start my PC in my local network.

**Reply by seriousm4x** [reply]
> How do I authorize and/or send a token?

Take a look at the [wiki](https://github.com/seriousm4x/UpSnap/wiki/Rest-API#authentication). You send a login request with your user and password as json body to either the admin or user route and get a bearer token back. You can use this token in the request header for all further requests.

> Can't I just disable that somehow? I just want to use a link to start my PC in my local network.

Also explained in the [wiki](https://github.com/seriousm4x/UpSnap/wiki/How-to-use-authorization#can-authorization-be-disabled).

**Reply by JonasHRR** [reply]
I saw the wiki, but my question is more about how to actually send that login request. If I understand correcty, I have to open localhost:8090/api/admins/auth-with-password, but even that request requires an authentification token. Could you answer in more details about what to do exactly? Thank you

**Reply by seriousm4x** [reply]
### 1. Get the token like this

```bash
curl -X POST -H "Content-Type: application/json" --data '{"identity": "your@email.com", "password": "yourpassword"}' "http://localhost:8090/api/admins/auth-with-password"
```

Response:

```json
{
  "admin": {
    "id": "k4idmnb7qq1lor7",
    "created": "2023-03-26 09:49:30.391Z",
    "updated": "2024-01-26 23:37:27.854Z",
    "avatar": 3,
    "email": "your@email.com"
  },
  "token": "YOUR-TOKEN"
}
```

### 2. Wake a device by id:

```bash
curl -X GET -H "Authorization: Bearer YOUR-TOKEN" "http://localhost:8090/api/upsnap/wake/xbgv4vruficbf31"
```

Response:

```json
{
  "collectionId": "z5lghx2r3tm45n1",
  "collectionName": "devices",
  "created": "2023-10-17 21:29:07.911Z",
  "created_by": "",
  "groups": [],
  "id": "xbgv4vruficbf31",
  "ip": "192.168.0.1",
  "link": "",
  "mac": "aa:aa:aa:aa:aa:aa",
  "name": "dummy",
  "netmask": "255.255.255.255",
  "password": "",
  "ping_cmd": "",
  "ports": [],
  "shutdown_cmd": "",
  "shutdown_confirm": false,
  "shutdown_cron": "",
  "shutdown_cron_enabled": false,
  "sol_auth": false,
  "sol_enabled": false,
  "sol_password": "",
  "sol_port": 0,
  "sol_user": "",
  "status": "pending",
  "updated": "2024-03-14 17:08:41.609Z",
  "wake_cmd": "",
  "wake_confirm": false,
  "wake_cron": "",
  "wake_cron_enabled": false
}
```

**Comment by Meth0d007** [comment]
Could you give an example of a simple wakeup request that can betriggered from a Windows machine from batch/ps (board tools)?
